build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,289 @@
+ * @throws If api-extractor config cannot be loaded.
+ */
+export function getTypeRollupPathFromExtractorConfig(
+	rollupType: "alpha" | "beta" | "public" | "untrimmed",
+	extractorConfig: IExtractorConfigPrepareOptions,
+): string | undefined {
+	try {
+		if (!extractorConfig || !extractorConfig.configObject) {
+			console.log("API Extractor configuration not found. Falling back to default behavior.");
+			return undefined;
+		}
+		const apiExtractorConfig = extractorConfig.configObject;
+		// Get rollupPath based on release tag
+		// https://api-extractor.com/pages/setup/configure_rollup/#trimming-based-on-release-tags
+		if (apiExtractorConfig.dtsRollup) {
+			let rollupPath: string | undefined;
+			if (rollupType === "untrimmed") {
+				rollupPath = apiExtractorConfig.dtsRollup.untrimmedFilePath;
+			} else {
+				rollupPath = apiExtractorConfig.dtsRollup[`${rollupType}TrimmedFilePath`];
+			}
+			if (!rollupPath) {
+				console.warn(`Rollup path for "${rollupType}" not found.`);
+				return undefined;
+			}
+			return rollupPath;
+		} else {
+			console.warn(`dtsRollup configuration not found in the API Extractor configuration.`);
+			return undefined;
+		}
+	} catch (error) {
+		console.error("Error loading API Extractor configuration:", error);
+		throw error;
+	}
+}
+
+/**
+ * Attempts to extract the type definition file path from the 'exports' field of a given package.json.
+ * Checks both 'import' and 'require' resolution methods to find the appropriate path.
+ * If the path is found, it is returned. Otherwise, an error is thrown.
+ * @param previousPackageJson - An object representing the previous package json
+ * @param previousBasePath - A string representing the path to the root of a package
+ * @returns A type definition filepath based on the appropriate export, or undefined if it cannot be found.
+ */
+export function getTypePathFromExport(
+	previousPackageJson: PackageJson,
+	previousBasePath: string,
+): string | undefined {
+	if (!previousPackageJson.exports) {
+		console.warn("The 'exports' field is missing in the package.json.");
+		return undefined;
+	}
+
+	const extractTypesPath = (exportEntry: any): string | undefined => {
+		return exportEntry?.types ? path.join(previousBasePath, exportEntry.types) : undefined;
+	};
+	const defaultSubpath = ".";
+	const exportEntry = previousPackageJson.exports[defaultSubpath];
+
+	// Check both 'import' and 'require' resolution methods
+	const typeDefinitionFilePath =
+		extractTypesPath(exportEntry?.import) ?? extractTypesPath(exportEntry?.require);
+	if (!typeDefinitionFilePath) {
+		// If no valid path is found, throw an error
+		throw new Error(
+			`Type definition file path could not be determined from the 'exports' field of '${getPreviousPackageJsonPath(
+				previousBasePath,
+			)}' using the default export entry '.'`,
+		);
+	}
+	return typeDefinitionFilePath;
+}
+/**
+ * Checks the package.json's exports entries and types field for a type definition filepath
```

Comment:
Nit, to emphasize the references to package.json members
```suggestion
 * Checks the package.json's `exports` entries and `types` field for a type definition file path.
```